input/tar2/tar_input.go

```diff
@@ -0,0 +1,163 @@
+package tar2
+
+import (
+	"archive/tar"
+	"bytes"
+	"crypto/sha512"
+	"encoding/base64"
+	"hash"
+	"io"
+	"os"
+	"path"
```

Comment:
Spec: don't care, that's what the `path.Clean` is supposed to handle.

This *isn't* file handling code.  The test code using `filepath` *is* file handling code.

Do you think we're going to improve readability by commenting every single use of `path` and `filepath` with a per-line "yes, author knew what was doing" signoff comment?  There are a lot of places that deal with files, and a lot of places that deal with canonicalized data.  That'll generate... a lot of comments.